percona/controllers/pgc/pgc.go

```diff
@@ -45,6 +50,51 @@ const (
 	defaultSecurityContext = `{"fsGroup": 26,"supplementalGroups": [1001]}`
 )
 
+type CronRegistry struct {
+	crons             *cron.Cron
+	ensureVersionJobs map[string]Schedule
+}
+
+type Schedule struct {
+	ID           int
+	CronSchedule string
+}
+
+func NewCronRegistry() CronRegistry {
+	c := CronRegistry{
+		crons:             cron.New(),
+		ensureVersionJobs: make(map[string]Schedule),
+	}
+
+	c.crons.Start()
+
+	return c
+}
+
+type lockStore struct {
+	store *sync.Map
+}
+
+func newLockStore() lockStore {
+	return lockStore{
+		store: new(sync.Map),
+	}
+}
+
+func (l lockStore) LoadOrCreate(key string) lock {
+	val, _ := l.store.LoadOrStore(key, lock{
+		statusMutex: new(sync.Mutex),
+		updateSync:  new(int32),
+	})
+
+	return val.(lock)
+}
+
+type lock struct {
+	statusMutex *sync.Mutex
+	updateSync  *int32
```

Comment:
we don't use this var anywhere, should we do something like https://github.com/percona/percona-xtradb-cluster-operator/blob/main/pkg/controller/pxc/controller.go#L201 in `c.onUpdate`?

Reply:
we also need to use it in `c.crons.crons.AddFunc` like in PXC: https://github.com/percona/percona-xtradb-cluster-operator/blob/main/pkg/controller/pxc/version.go#L70-L72

Reply:
https://github.com/percona/percona-postgresql-operator/pull/152/commits/ea0b9ffd075cabc55400a93a80a335750e274fcd